### PR TITLE
enable support for float arguments

### PR DESCRIPTION
**Pull request by pariasm**
A small modification to allow floats as program arguments.

### CODE REVIEW

File: test_argparse.c [hunk 1]
```diff
@@ -25,6 +26,7 @@ main(int argc, const char **argv)
         OPT_BOOLEAN('t', "test", &test, "test only"),
         OPT_STRING('p', "path", &path, "path to read"),
         OPT_INTEGER('n', "num", &num, "selected num"),
```
Comment (mschilli87) [inline test_argparse.c]: Shouldn't this be re-named to `"int"` or sth. now? The choice `"num"`/`"float"` seems a bit confusing now that there are several numeric options.

Comment (pariasm) [reply test_argparse.c]: You're right. I'll change that and I'll also update the unit tests. I'll submit another pull request with these changes.

Comment (pariasm) [reply test_argparse.c]: Done. It seems that I don't need to create another pull request. My latest commit appears in this one.

**Comment by cofyc** [comment]
This is great! Thanks for the contribution!

**Pull request merged**